AppHandling/Start-NavContainerAppDataUpgrade.ps1

```diff
@@ -18,39 +18,64 @@ function  Start-BcContainerAppDataUpgrade {
-        [Parameter(Mandatory=$true)]
+        [Parameter(Mandatory = $true)]
         [string] $appName,
         [string] $appVersion = "",
-        [string] $language = ""
+        [string] $language = "",
+        [string] $exclusiveAccessTicket = "",
+        [string] $path = "",
+        [switch] $force,
+        [switch] $skipVersionCheck,
+        [ValidateSet("Add", "Clean", "Development", "ForceSync", "None")]
+        [string] $syncMode = ""
     )
 
-$telemetryScope = InitTelemetryScope -name $MyInvocation.InvocationName -parameterValues $PSBoundParameters -includeParameters @()
-try {
-
-    Invoke-ScriptInBcContainer -containerName $containerName -ScriptBlock { Param($appName, $appVersion, $tenant, $language)
-        Write-Host "Upgrading app $appName"
-        $parameters = @{
-            "ServerInstance" = $ServerInstance;
-            "Name" = $appName;
-            "Tenant" = $tenant
-        }
-        if ($appVersion) {
-            $parameters += @{ "Version" = $appVersion }
+    $telemetryScope = InitTelemetryScope -name $MyInvocation.InvocationName -parameterValues $PSBoundParameters -includeParameters @()
+    try {
+        $containerPath = = ""
+        if ($path) {
+            $containerPath = (Get-BcContainerPath -containerName $containerName -path $path -throw)
         }
-        if ($language) {
-            $parameters += @{ "Language" = $language }
-        }
-        Start-NAVAppDataUpgrade @parameters
-    } -ArgumentList $appName, $appVersion, $tenant, $language
-    Write-Host -ForegroundColor Green "App successfully upgraded"
-}
-catch {
-    TrackException -telemetryScope $telemetryScope -errorRecord $_
-    throw
-}
-finally {
-    TrackTrace -telemetryScope $telemetryScope
-}
+
+        Invoke-ScriptInBcContainer -containerName $containerName -ScriptBlock { Param($appName, $appVersion, $tenant, $language, $exclusiveAccessTicket, $containerPath, $syncMode, $force, $skipVersionCheck)
+            Write-Host "Upgrading app $appName"
+            $parameters = @{
+                "ServerInstance" = $ServerInstance;
+                "Name"           = $appName;
+                "Tenant"         = $tenant
+            }
+            if ($appVersion) {
+                $parameters += @{ "Version" = $appVersion }
+            }
+            if ($language) {
+                $parameters += @{ "Language" = $language }
+            }
+            if ($exclusiveAccessTicket) {
+                $parameters += @{ "ExclusiveAccessTicket" = $exclusiveAccessTicket }
+            }
+            if ($containerPath) {
+                $parameters += @{ "Path" = $containerPath }
+            }
+            if ($syncMode) {
+                $parameters += @{ "SyncMode" = $syncMode }
+            }
+            if ($force.IsPresent) {
+                $parameters += @{ "Force" = $force }
+            }
+            if ($skipVersionCheck.IsPresent) {
+                $parameters += @{ "SkipVersionCheck" = $skipVersionCheck }

```

Comment:
I think these needs to be 
"skipVersionCheck" = $true
(same with force and exclusiveAccessTicket)

Reply:
-exclusiveAccessTicket is string value 
-force and -skipVersionCheck are in default set to $false. It means that your function works now with these defaults. Do you want change defaults? It going to change function working for all who use this function without new paramaters.

Reply:
The only change I want is this:

```
if ($force.IsPresent) {
    $parameters += @{ "Force" = $true }
}
if ($skipVersionCheck.IsPresent) {
    $parameters += @{ "SkipVersionCheck" = $true}
}
```

mostly because force is of type System.Management.Automation.SwitchParameter and I am always in doubt whether the other construct works...
It does - I tried, but to improve readability...

Reply:
I catch the point now :-). Sorry for my misunderstanding. I changed it.